x-pack/plugins/apm/public/components/app/Main/routeConfig.tsx

```diff
@@ -25,6 +25,25 @@ interface RouteParams {
   serviceName: string;
 }
 
+type BreadcrumbFunction = (args: BreadcrumbArgs) => string | null;
+
+interface Route extends RouteProps {
+  switch?: never;
```

Comment:
i'm not sure this is the intention of the `never` type. should it be `boolean` instead?

Reply:
I think you can simplify this if you combine the `switch` and `routes` properties into just a `switchRoutes` (renamed for clarity). Then you can type it like:

```ts
interface Route extends RouteProps {
  switchRoutes?: Route[];
  breadcrumb?: string | BreadcrumbFunction | null;
}

type RoutesConfig = Route[];
```

If you want the union you can do:
```ts
interface NormalRoute extends RouteProps {
  breadcrumb: string | BreadcrumbFunction | null;
}

interface SwitchRoute extends RouteProps {
  switchRoutes: Route[];
}

type Route = SwitchRoute | NormalRoute;
```

But I think the extra type narrowing is not warranted.

Reply:
If you prefer the second solution, we can try working towards that. What do you mean "it doesn't work"?

> If we're ok with changing the key to switchRoutes and just checking for its existence, I'm good with that.

Yes, I think the is the better solution, Typescript or not. Just didn't think about it back when I initially wrote it

Reply:
I don't really care that much. :) I made the change to `switchRoutes` :+1: